src/ir/node_types.go

```diff
@@ -531,7 +531,7 @@ type InstanceOfExpr struct {
 	FreeFloating freefloating.Collection
 	Position     *position.Position
 	Expr         Node
-	Class        Node
+	Class        Node // *ir.Identifier (static), *ir.Name (ClassNames, self), *ir.SimpleVar ($this)
```

Comment:
I think `Class` operand of the instance of can be literally anything.
A function call, index expression, etc.